server/service/src/preference/upsert.rs

```diff
@@ -0,0 +1,38 @@
+use repository::{PreferenceRow, PreferenceRowRepository, RepositoryError};
+
+use crate::service_provider::ServiceContext;
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct UpsertPreference {
+    pub id: String,
+    pub store_id: Option<String>,
+    pub key: String,
+    pub value: String,
+}
+
+pub fn upsert_preference(
```

Comment:
I wanted this to be type safe, and not at runtime but at compile time. 

The thought that I had was having upsert method for every preference, similar to the query method in Preference for now. And when we do by store preference or machine uuid we would probably need a different type of save, maybe with array of store -> pref objects.